pkg/slack/slack.go

```diff
@@ -56,12 +56,42 @@ func (b *Bot) Start() {
 			logging.Logger.Debug("Connection Info: ", ev.Info)
 
 		case *slack.MessageEvent:
-			// Serve only if mentioned
-			if !strings.HasPrefix(ev.Text, "<@"+botID+"> ") {
+			// Check if message posted in a channel
+			chanInfo, _ := api.GetChannelInfo(ev.Channel)
+
+			// Check if message posted in a group
+			groupInfo, _ := api.GetGroupInfo(ev.Channel)
+
+			if chanInfo != nil || groupInfo != nil {
+				// Message posted in a channel/group
+				// Serve only if starts with mention
+				if !strings.HasPrefix(ev.Text, "<@"+botID+"> ") {
+					continue
+				}
+				logging.Logger.Debugf("Slack incoming message: %+v", ev)
+				msg := strings.TrimPrefix(ev.Text, "<@"+botID+"> ")
+				sm := slackMessage{
+					ChannelID: ev.Channel,
+					BotID:     botID,
+					InMessage: msg,
+					RTM:       rtm,
+				}
+				sm.HandleMessage(b.AllowKubectl)
+				continue
+			}
+
+			// Message posted as a DM
+			// Skip if message posted by BotKube
+			if ev.User == botID {
 				continue
 			}
 			logging.Logger.Debugf("Slack incoming message: %+v", ev)
-			msg := strings.TrimPrefix(ev.Text, "<@"+botID+"> ")
+			msg := ev.Text
+
+			// Trim @BotKube prefix if exists
+			if strings.HasPrefix(ev.Text, "<@"+botID+"> ") {
```

Comment:
this seems to be correct place to trim message. Rest are redundant and logic can be arranged so that code arrives here.